src/Access/RoleCache.cpp

```diff
@@ -156,17 +168,46 @@ RolePtr RoleCache::getRole(const UUID & role_id)
 }
 
 
+RolePtr RoleCache::getRole(const UUID & role_id)
+{
+    /// `mutex` is already locked.
+
+    LOG_TRACE(&Poco::Logger::get("RoleCache ()"), "top of getRole");
+    auto role_from_cache = cache.get(role_id);
+    if (role_from_cache)
+        return role_from_cache->first;
+
+    return mkRoleEntry(role_id);
+}
+
+
 void RoleCache::roleChanged(const UUID & role_id, const RolePtr & changed_role)
 {
+    LOG_TRACE(&Poco::Logger::get("RoleCache ()"), "top of roleChanged");
```

Comment:
Minor: Unless this is a temporary thing, to be removed before merging, consider putting a logger into a separate function, like
```namespace
{
auto * getLogger()
{
    static Poco::Logger & logger = Poco::Logger::get("RoleCache");
    return &logger;
}
}
```

or put it as member into `RoleCache`?

Reply:
It is temporary of course.
I thought that draft PR is an excuse to keep debugging stuff.

Reply:
sure, this is just a friendly reminder